tests/unit/backend/recipe/algorithm/test_PixelDiffractionCalibration.py

```diff
@@ -1,153 +1,50 @@
 
 import pytest
-from mantid.simpleapi import (
-    DeleteWorkspace,
-    mtd,
-)
+from mantid.api import MatrixWorkspace
+from mantid.dataobjects import MaskWorkspace
+from mantid.simpleapi import mtd
 from snapred.backend.dao.DetectorPeak import DetectorPeak
 from snapred.backend.dao.GroupPeakList import GroupPeakList
 from snapred.backend.dao.ingredients import DiffractionCalibrationIngredients
-
-# needed to make mocked ingredients
-from snapred.backend.dao.RunConfig import RunConfig
-from snapred.backend.dao.state.FocusGroup import FocusGroup
-from snapred.backend.dao.state.InstrumentState import InstrumentState
 from snapred.backend.dao.state.PixelGroup import PixelGroup
 
 # the algorithm to test
 from snapred.backend.recipe.algorithm.PixelDiffractionCalibration import (
     PixelDiffractionCalibration as ThisAlgo,  # noqa: E402
 )
 from snapred.meta.Config import Resource
+from util.diffraction_calibration_synthetic_data import SyntheticData
+from util.helpers import deleteWorkspaceNoThrow, maskSpectra, setSpectraToZero
 
 
 class TestPixelDiffractionCalibration(unittest.TestCase):
     def setUp(self):
         """Create a set of mocked ingredients for calculating DIFC corrected by offsets"""
-        self.maxOffset = 2
-        self.fakeRunNumber = "555"
-        fakeRunConfig = RunConfig(runNumber=str(self.fakeRunNumber))
-
-        fakeInstrumentState = InstrumentState.parse_raw(Resource.read("/inputs/diffcal/fakeInstrumentState.json"))
-
-        fakeFocusGroup = FocusGroup.parse_raw(Resource.read("/inputs/diffcal/fakeFocusGroup.json"))
-        fakeFocusGroup.definition = Resource.getPath("inputs/testInstrument/fakeSNAPFocGroup_Natural.xml")
-
-        peakLists = {
-            3: DetectorPeak.parse_obj({"position": {"value": 0.2, "minimum": 0.15, "maximum": 0.25}}),
-            7: DetectorPeak.parse_obj({"position": {"value": 0.3, "minimum": 0.20, "maximum": 0.40}}),
-            2: DetectorPeak.parse_obj({"position": {"value": 0.18, "minimum": 0.10, "maximum": 0.25}}),
-            11: DetectorPeak.parse_obj({"position": {"value": 0.24, "minimum": 0.18, "maximum": 0.30}}),
-        }
-        self.fakeIngredients = DiffractionCalibrationIngredients(
-            runConfig=fakeRunConfig,
-            focusGroup=fakeFocusGroup,
-            instrumentState=fakeInstrumentState,
-            groupedPeakLists=[
-                GroupPeakList(groupID=gid, peaks=[peakLists[gid]], maxfwhm=5) for gid in peakLists.keys()
-            ],
-            calPath=Resource.getPath("outputs/calibration/"),
-            convergenceThreshold=1.0,
-            maxOffset=self.maxOffset,
-            pixelGroup=fakeInstrumentState.pixelGroup,
-        )
-
-        self.fakeRawData = f"_test_pixelcal_{self.fakeRunNumber}"
-        self.fakeGroupingWorkspace = f"_test_pixelcal_difc_{self.fakeRunNumber}"
-        self.makeFakeNeutronData(self.fakeRawData, self.fakeGroupingWorkspace)
+        inputs = SyntheticData()
```

Comment:
Should this be in `setUpClass`?

Reply:
**A key issue is that these tests are designed to be successfully run in parallel.**  We don't do this yet, but it's one of the better ways to reduce our test suite execution time.  Note that there are two versions of how this data is used:  (1) `test_PixelDiffractionCalibration` (here), and (2) `test_GroupDiffractionCalibration`, which does use `setUpClass` and needs to _clone_ the input data in many of the tests.  
It was beyond the scope of this story to attempt to _unify_ the entire diffraction-calibration test suite.  :)